### PR TITLE
docs(gke-tpu): add walkthrough for using sglang-jax on GKE with TPUs

### LINKED ISSUE #869: [Docs]: Add Docker and Kubernetes installation guides
### Checklist

- [x] 1. Make sure you already searched for relevant issues and documentations.
- [x] 2. Please use English, otherwise it will be closed.

### Documentation Issues or Suggested Changes

  The installation guide [docs/get_started/install.md](https://github.com/sgl-project/sglang-jax/blob/main/docs/get_started/install.md) currently has placeholder sections for Method 3 (Docker) and Method 4 (Kubernetes), both marked as 🚧 **Under Construction** 🚧.

  We should add complete instructions for these two deployment methods

### PR DESCRIPTION
**Pull request by aaron-prindle**
## Motivation
Fixes https://github.com/sgl-project/sglang-jax/issues/869

Completes one possible Kubernetes deployment path for `sglang-jax` (previously marked as "Under Construction" in the docs) using GKE + TPUs. Serving JAX on GKE TPUs has several gotchas like rolling update deadlocks and probe JIT cancellations. This PR adds a step-by-step GKE TPU guide as a quickstart for users.

NOTE: Can definitely rewrite the `## Method 4: Using Kubernetes` section a bit if we want to split out GKE from other K8s usages. Currently wrote it as the GKE + TPU method fulfilling `Method 4` entirely.

## Modifications

1. **docs/get_started/gke-tpu-install.md** (New):
   - Setup guides for GKE cluster, TPU node pool (`ct5lp-hightpu-8t`), registry, and IAM.
   - Verified k8s Deployment/Service manifest with JAX/TPU optimizations (`Recreate` strategy, `python -u` logs, TP/DP mesh tuning, robust timeouts).
   - Friction points section documenting GKE deadlocks, JIT cancellations, `--tp-size` world size gotcha, and watchdog timeouts.
   - Cleanup instructions.

2. **docs/get_started/install.md** (Modified):
   - Replaced "Under Construction" placeholder with a link to the new guide.

## Accuracy Tests

Verified with `Qwen/Qwen2.5-7B-Instruct` on a GKE TPU v5e-8 node. The model served successfully using `--tp-size 8 --dp-size 2` (actual TP=4, DP=2) and returned correct chat completions:

```bash
curl http://localhost:30000/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{"model": "Qwen/Qwen2.5-7B-Instruct", "messages": [{"role": "user", "content": "Hello, tell me a joke."}]}'
```
- Got expected response -  "Why don't scientists trust atoms? Because they make up everything!"

## Benchmarking and Profiling

Documentation-only, no performance impact. Bypassed the default 30+ min precompilation delay by recommending `--disable-precompile` for dev, paired with adjusted `--watchdog-timeout 1200` and `SGLANG_WAIT_TIMEOUT=600` to prevent warmup timeouts.

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

File: docs/get_started/gke-tpu-install.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+This guide provides step-by-step instructions for deploying and serving Large Language Models (LLMs) using **SGLang-Jax** on **Google Kubernetes Engine (GKE)** equipped with **Cloud TPUs**. JAX's unique JIT compilation model requires specific configurations to avoid common pitfalls like startup timeouts, deadlocks, etc. so this guide has the correct configurations for a good getting started experience.
+
+> [!IMPORTANT]
+> This guide is designed to help you get familiar with SGLang-Jax on GKE TPU in a development environment. The configurations presented here (such as disabling JIT precompilation and using lightweight `/health` checks) are optimized for fast feedback and quick testing, and are **not recommended for production deployments**.
+
+## Prerequisites
+
+1.  **Google Cloud Project**: An active GCP project (eg: `your-project-id`).
+2.  **TPU Quota**: Ensure you have quota for Cloud TPU v5e (specifically `ct5lp-hightpu-8t` machine type) in your chosen region (eg: `us-west1-c`).
```
Comment (zhengkezhou1) [inline docs/get_started/gke-tpu-install.md]: We shouldn't make assumptions about users running a specific TPU machine type in a particular region.
Consider making this more generic, or framing it as an example rather than a requirement.

File: docs/get_started/gke-tpu-install.md [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+3.  **gcloud CLI & kubectl**: Installed and authenticated.
+
+
+## Step 1: Create a GKE Cluster and TPU Node Pool
+
+First, set up your GKE cluster and provision a dedicated Cloud TPU v5e node pool.
```
Comment (zhengkezhou1) [inline docs/get_started/gke-tpu-install.md]: above

**Review by zhengkezhou1 — COMMENTED** [review]
we're actively working on the sgl-jax release pipeline. Once it lands, users will be able to pull the image directly from Docker Hub, which should let us drop a number of these prerequisite steps.

File: docs/get_started/gke-tpu-install.md (hunk 1 above)
Comment (aaron-prindle) [reply docs/get_started/gke-tpu-install.md]: +1, makes sense. Addressed now, made this more generic by:

- Updating the prerequisites to frame the Cloud TPU v5e (`ct5lp-hightpu-8t`) and `us-west1-c` region as a concrete example rather than a strict requirement.
- Introducing a generic `MACHINE_TYPE` and `NODE_POOL_NAME` variable in the environment setup step, and updating the node pool creation commands to use them dynamically.
- Adding inline "NOTE:" comments inside the Kubernetes YAML manifest to guide users on how to adapt the image path, `nodeSelector`, and `resources` limits/requests to match their specific setup.
- Updating the manifest introduction note to guide users to look for these "NOTE:" comments in the YAML.
- Framed the specific TPU v5e math in the "Friction Points" section as an illustrative example.

File: docs/get_started/gke-tpu-install.md (hunk 2 above)
Comment (aaron-prindle) [reply docs/get_started/gke-tpu-install.md]: Done. This is now fully parameterized using the new `MACHINE_TYPE` and `NODE_POOL_NAME` environment variables.

**Comment by aaron-prindle** [comment]
> we're actively working on the sgl-jax release pipeline. Once it lands, users will be able to pull the image directly from Docker Hub, which should let us drop a number of these prerequisite steps.

Oh nice! I added small blurb noting this so folks know they can check for it:
```
*(Note: The sgl-jax team is actively working on an official release pipeline. Once it lands, you will be able to pull the official image directly from Docker Hub and skip this build step).*
```

**Comment by aaron-prindle** [comment]
@zhengkezhou1  - addressed your  initial comments there.  PTAL when you get a chance, thanks!

**Comment by zhengkezhou1** [comment]
> > we're actively working on the sgl-jax release pipeline. Once it lands, users will be able to pull the image directly from Docker Hub, which should let us drop a number of these prerequisite steps.
> 
> Oh nice! I added small blurb noting this so folks know they can check for it:
> 
> ```
> *(Note: The sgl-jax team is actively working on an official release pipeline. Once it lands, you will be able to pull the official image directly from Docker Hub and skip this build step).*
> ```

Just a quick heads-up: we actually already have images available at https://hub.docker.com/r/lmsysorg/sglang-jax.

Even though the full release pipeline is still in progress, we can update the note to point users to this existing image for now.

**Comment by aaron-prindle** [comment]
> Just a quick heads-up: we actually already have images available at https://hub.docker.com/r/lmsysorg/sglang-jax.

> Even though the full release pipeline is still in progress, we can update the note to point users to this existing image for now.

Great, updated the docs here now to remove the build section, having to use a container registry for the build (for GCP this was Artifact Registry) and removed needed to setup k8s cluster perms to access the container registry the user previously pushed to.  This has simplified the docs here now.

**Comment by aaron-prindle** [comment]
@zhengkezhou1  - addressed your comments related to the container existing now (nice!). PTAL when you get a chance, thanks!

File: docs/get_started/gke-tpu-install.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    metadata:
+      labels:
+        app: sglang-tpu
+    spec:
+      nodeSelector:
+        # NOTE: The nodeSelector below is configured for TPU v5e (2x4 topology).
+        # Adjust these labels to match your allocated TPU type and topology (eg: v4, v6e).
+        cloud.google.com/gke-tpu-accelerator: tpu-v5-lite-podslice
+        cloud.google.com/gke-tpu-topology: 2x4
+        cloud.google.com/gke-spot: "true"
```
Comment (zhengkezhou1) [inline docs/get_started/gke-tpu-install.md:94]: Since the example already creates a Spot node pool by default, `cloud.google.com/gke-spot: "true"` does not affect scheduling behavior in the current setup. It would only be necessary when the cluster contains both Spot and non-Spot node pools, or when we want to explicitly constrain workloads to Spot nodes for future-proofing.

**Review by zhengkezhou1 — COMMENTED** [review]
Just one minor comment left. Overall, this looks very good!

**Comment by zhengkezhou1** [comment]
@aaron-prindle There are still a few minor issues, but nothing blocking, so I went ahead and merged this.

**Review by zhengkezhou1 — APPROVED** [review]
